### PR TITLE
Service client message undispose

### PR DESCRIPTION
**Pull request by tmahmood-microsoft**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `main` branch.

### CODE REVIEW

File: e2e/test/iothub/service/IoTHubServiceProxyE2ETests.cs [hunk 1]
```diff
@@ -50,6 +50,7 @@ public async Task RegistryManager_AddAndRemoveDevice_WithProxy()
         }
 
         [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
+        [Ignore]
```
Comment (drwill-ms) [inline e2e/test/iothub/service/IoTHubServiceProxyE2ETests.cs:53]: Was this change intended?

File: e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace Microsoft.Azure.Devices.E2ETests.iothub.service
+{
+    /// <summary>
+    /// E2E test class for PurgeMesageQueue.
+    /// </summary>
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public class PurgeMesageQueueE2ETests : E2EMsTestBase
```
Comment (drwill-ms) [inline e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs:15]:
```suggestion
    public class PurgeMesageQueueE2eTests : E2EMsTestBase
```

File: e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                DeviceId = deviceId,
+                TotalMessagesPurged = 50
+            };
+            for (int i = 0; i < 50; ++i)
+            {
+                await sc.Messaging.SendAsync(deviceId, testMessage);
+            }
+            PurgeMessageQueueResult result = await sc.Messaging.PurgeMessageQueueAsync(deviceId, CancellationToken.None).ConfigureAwait(false);
+            Assert.AreEqual(expectedResult.DeviceId, result.DeviceId);
+            Assert.AreEqual(expectedResult.TotalMessagesPurged, result.TotalMessagesPurged);
```
Comment (drwill-ms) [inline e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs:34]: Please take the opportunity to change this test to use fluent assertions.

```suggestion
            result.DeviceId.Should().Be(expectedResult.DeviceId);
            result.TotalMessagesPurged.Should().Be(expectedResult.TotalMessagesPurged);
```

File: e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    [TestCategory("IoTHub")]
+    public class PurgeMesageQueueE2ETests : E2EMsTestBase
+    {
+        [TestMethod]
+        public async Task PurgeMessageQueueOperation()
+        {
+            Message testMessage = ComposeD2CTestMessage();
+            using var sc = new IotHubServiceClient(TestConfiguration.IoTHub.ConnectionString);
+            var deviceId = TestConfiguration.IoTHub.X509ChainDeviceName;
+            var expectedResult = new PurgeMessageQueueResult()
```
Comment (drwill-ms) [inline e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs:23]: nit: when using an initializer list and no ctor parameters, you can drop the parentheses.
```suggestion
            var expectedResult = new PurgeMessageQueueResult
```

File: e2e/test/iothub/service/IoTHubServiceProxyE2ETests.cs (hunk 1 above)
Comment (tmahmood-microsoft) [reply e2e/test/iothub/service/IoTHubServiceProxyE2ETests.cs:53]: https://msazure.visualstudio.com/One/_workitems/edit/15302262

Added a task to look into it later.
This test is failing with timeout issue.
